modules/Users/Changenewpassword.php

```diff
@@ -98,166 +86,126 @@
+    $Q = "SELECT * FROM users_password_link WHERE id = '" . $_REQUEST['guid'] . "' AND COALESCE(deleted, 0) = '0'";
+    $result = $GLOBALS['db']->limitQuery($Q, 0, 1, false);
+    $row = $GLOBALS['db']->fetchByAssoc($result);
+    if (!empty($row)) {
+        $pwd_settings = $GLOBALS['sugar_config']['passwordsetting'];
+        $expired = '0';
+        if ($pwd_settings['linkexpiration']) {
+            $delay = $pwd_settings['linkexpirationtime'] * $pwd_settings['linkexpirationtype'];
             $stim = strtotime($row['date_generated']) + date('Z');
-			$expiretime = TimeDate::getInstance()->fromTimestamp($stim)->get("+$delay  minutes")->asDb();
-	    	$timenow = TimeDate::getInstance()->nowDb();
-	    	if ($timenow > $expiretime)
-	    		$expired='1';
-	    }
-
-	    if (!$expired)
-	    	{
-	    		// if the form is filled and we want to login
-	    		if (isset($_REQUEST['login']) && $_REQUEST['login'] =='1'){
-	    			if ( $row['username'] == $_POST['user_name'] ){
-
-						$usr= new user();
-						$usr_id=$usr->retrieve_user_id($_POST['user_name']);
-	    				$usr->retrieve($usr_id);
-	    				$usr->setNewPassword($_POST['new_password']);
-					    $query2 = "UPDATE users_password_link SET deleted='1' where id='".$_REQUEST['guid']."'";
-				   		$GLOBALS['db']->query($query2, true, "Error setting link for $usr->user_name: ");
-				   		$_POST['user_name'] = $_REQUEST['user_name'];
-						$_POST['username_password'] = $_REQUEST['new_password'];
-						$_POST['module'] = 'Users';
-						$_POST['action'] = 'Authenticate';
-						$_POST['login_module'] = 'Home';
-						$_POST['login_action'] = 'index';
-						$_POST['Login'] = 'Login';
-						foreach($_POST as $k=>$v){
-							$_REQUEST[$k] = $v;
-							$_GET[$k]= $v;
-						}
-						unset($_REQUEST['entryPoint']);
-						unset($_GET['entryPoint']);
-						$GLOBALS['app']->execute();
-						die();
-				   	}
-	    		}
-				else
-				$redirect='0';
-    		}
-		else
-			{
-				$query2 = "UPDATE users_password_link SET deleted='1' where id='".$_REQUEST['guid']."'";
-		    	$GLOBALS['db']->query($query2, true, "Error setting link");
-			}
- 		}
- 	}
+            $expiretime = TimeDate::getInstance()->fromTimestamp($stim)->get("+$delay  minutes")->asDb();
+            $timenow = TimeDate::getInstance()->nowDb();
+            if ($timenow > $expiretime) {
+                $expired = '1';
+            }
+        }
+
+        if (!$expired) {
+            // if the form is filled and we want to login
+            if (isset($_REQUEST['login']) && $_REQUEST['login'] == '1') {
+                if ($row['username'] == $_POST['user_name']) {
+
+                    $usr = new user();
+                    $usr_id = $usr->retrieve_user_id($_POST['user_name']);
+                    $usr->retrieve($usr_id);
+                    $usr->setNewPassword($_POST['new_password']);
+                    $query2 = "UPDATE users_password_link SET deleted='1' where id='" . $_REQUEST['guid'] . "'";
```

Comment:
please make sure, the queries are secure, parameters prepared and escaped correctly.

Reply:
I have add ```$db->quote($_REQUEST['guid'])``` to correct the issue.